service/src/test/java/bio/terra/catalog/datarepo/DatarepoServiceTest.java

```diff
@@ -71,29 +72,27 @@ void getSnapshotsException() throws Exception {
   }
 
   @Test
-  void userHasActionReader() throws Exception {
+  void getRoleReader() throws Exception {
     var id = UUID.randomUUID();
     when(snapshotsApi.retrieveUserSnapshotRoles(id))
         .thenReturn(List.of(DatarepoService.READER_ROLE_NAME));
-    assertTrue(datarepoService.userHasAction(user, id.toString(), SamAction.READ_ANY_METADATA));
+    assertEquals(datarepoService.getRole(user, id.toString()), DatasetAccessLevel.READER);
```

Comment:
I would use `assertThat()` here instead, since it's usually better at printing out differences between objects. If you do use `assertEquals()`, you should swap the argument order, since the expected value comes first.
```suggestion
    assertThat(datarepoService.getRole(user, id.toString()), is(DatasetAccessLevel.READER));
```